packages/block-library/src/query/edit/inspector-controls/index.js

```diff
@@ -69,64 +48,56 @@ export default function QueryInspectorControls( {
 		postType,
 		sticky,
 		inherit,
+		taxQuery,
 	} = query;
-	const [ showCategories, setShowCategories ] = useState( true );
-	const [ showTags, setShowTags ] = useState( true );
 	const [ showSticky, setShowSticky ] = useState( postType === 'post' );
 	const { postTypesTaxonomiesMap, postTypesSelectOptions } = usePostTypes();
-	const { categories, tags } = useSelect( ( select ) => {
-		const { getEntityRecords } = select( coreStore );
-		const termsQuery = { per_page: MAX_FETCHED_TERMS };
-		const _categories = getEntityRecords(
-			'taxonomy',
-			'category',
-			termsQuery
-		);
-		const _tags = getEntityRecords( 'taxonomy', 'post_tag', termsQuery );
-		return {
-			categories: getEntitiesInfo( _categories ),
-			tags: getEntitiesInfo( _tags ),
-		};
-	}, [] );
-	useEffect( () => {
-		if ( ! postTypesTaxonomiesMap ) return;
-		const postTypeTaxonomies = postTypesTaxonomiesMap[ postType ];
-		setShowCategories( postTypeTaxonomies.includes( 'category' ) );
-		setShowTags( postTypeTaxonomies.includes( 'post_tag' ) );
-	}, [ postType, postTypesTaxonomiesMap ] );
 	useEffect( () => {
 		setShowSticky( postType === 'post' );
 	}, [ postType ] );
+	const { __unstableMarkNextChangeAsNotPersistent } = useDispatch(
+		blockEditorStore
+	);
+	// We need to migrate `categoryIds` and `tagIds` to `tax_query`.
+	// TODO: We could probably use the `deprecations` API but it
+	// doesn't support dynamic blocks deprecations properly.
```

Comment:
In what way does it not support dynamic blocks very well? I have successfully used it in projects by manually including the `isEligible` check which could also be done here to check whether the `categoryIds` key is present.

Reply:
The current API is designed for static blocks. We still need to add deprecation logic inside the server side render function of a block.

Reply:
@ntsekouras Yeah that is true for the php rendering where we will always need to expect old non-updated data. But the JS migration of the attributes can already be handled via the deprecations api. This way you could prevent having to use the `__unstableMarkNextChangeAsNotPersistent` API

Reply:
I know. I just don't like this inconsistency and wanted to think a bit more about it(thus `todo` 😄 ). You're right though that we should use the API and remove the previous fields. I've updated now with deprecations API. Thanks!